velox/expression/CastExpr.cpp

```diff
@@ -917,6 +917,185 @@ void CastExpr::applyPeeled(
   }
 }
 
+namespace {
+
+void castIntegerToBigintScalar(
+    const SelectivityVector& rows,
+    const int32_t* in,
+    int64_t* out) {
+  vector_size_t n = rows.end();
+  vector_size_t i = rows.begin();
+  for (; i + 4 <= n; i += 4) {
+    out[i] = static_cast<int64_t>(in[i]);
+    out[i + 1] = static_cast<int64_t>(in[i + 1]);
+    out[i + 2] = static_cast<int64_t>(in[i + 2]);
+    out[i + 3] = static_cast<int64_t>(in[i + 3]);
+  }
+  for (; i < n; ++i) {
+    out[i] = static_cast<int64_t>(in[i]);
+  }
+}
+
+#if (defined(__x86_64__) || defined(_M_X64)) && defined(__AVX2__)
+// If compiled with -mavx2, we can directly call AVX2 intrinsics.
+inline void castIntegerToBigintAvx2(
```

Comment:
This is not needed, just do a for loop and compiler will use SIMD automatically: https://godbolt.org/z/P8EzYW1fG